File: backend/internal/models/community.go
package models

import "time"

type CommunitySummary struct {
	ReportsMade     int     `json:"reports_made"`
	WasteDivertedKg float64 `json:"waste_diverted_kg"`
	Rank            int     `json:"rank"`
	TotalReporters  int     `json:"total_reporters"`
}

type ActivityItem struct {
	Type    string    `json:"type"`
	Message string    `json:"message"`
	Time    time.Time `json:"time"`
}

// ScheduledCollection is a collection stop on a planned or active route that
// has not been collected yet. Related marks points the resident reported on.
type ScheduledCollection struct {
	WastePointID      string  `json:"waste_point_id"`
	WastePointName    string  `json:"waste_point_name"`
	Latitude          float64 `json:"latitude"`
	Longitude         float64 `json:"longitude"`
	CurrentLevelPct   int     `json:"current_level_pct"`
	Status            string  `json:"status"`
	RouteID           string  `json:"route_id"`
	RouteStatus       string  `json:"route_status"`
	Order             int     `json:"order"`
	StopCount         int     `json:"stop_count"`
	EstimatedMinutes  int     `json:"estimated_minutes"`
	TruckRegistration string  `json:"truck_registration"`
	Related           bool    `json:"related"`
}

type PastCollection struct {
	ID             string    `json:"id"`
	WastePointID   string    `json:"waste_point_id"`
	WastePointName string    `json:"waste_point_name"`
	Outcome        string    `json:"outcome"`
	EstimatedKg    float64   `json:"estimated_kg"`
	CollectedAt    time.Time `json:"collected_at"`
}

type CommunityCollections struct {
	Upcoming []ScheduledCollection `json:"upcoming"`
	Past     []PastCollection      `json:"past"`
}

File: frontend/src/types/community.js
/**
 * @typedef {Object} CommunitySummary
 * @property {number} reports_made
 * @property {number} waste_diverted_kg
 * @property {number} rank
 * @property {number} total_reporters
 *
 * @typedef {Object} ActivityItem
 * @property {'collection'|'report'} type
 * @property {string} message
 * @property {string} time
 *
 * @typedef {Object} ScheduledCollection
 * @property {string} waste_point_id
 * @property {string} waste_point_name
 * @property {number} latitude
 * @property {number} longitude
 * @property {number} current_level_pct
 * @property {import('./wastePoint').WastePoint['status']} status
 * @property {string} route_id
 * @property {import('./route').Route['status']} route_status
 * @property {number} order
 * @property {number} stop_count
 * @property {number} estimated_minutes
 * @property {string} truck_registration
 * @property {boolean} related
 *
 * @typedef {Object} PastCollection
 * @property {string} id
 * @property {string} waste_point_id
 * @property {string} waste_point_name
 * @property {'collected'|'failed'} outcome
 * @property {number} estimated_kg
 * @property {string} collected_at
 *
 * @typedef {Object} CommunityCollections
 * @property {ScheduledCollection[]} upcoming
 * @property {PastCollection[]} past
 */

export {}
